datamapper/changeset/changeset.py

```diff
@@ -49,11 +32,15 @@ def cast(self, params: Mapping[str, Any], permitted: List[str]) -> Changeset:
         """
         Applies `params` as changes to the changeset, provided that their keys are in `permitted` and their types are correct.
         """
-        permitted_params = {k: v for (k, v) in params.items() if k in permitted}
-        return self._update(
-            params={**self.params, **permitted_params},
-            permitted=self.permitted.union(permitted),
+        self.params = {k: v for (k, v) in params.items() if k in permitted}
+        (changes, errors) = validators.validate_types(
```

Comment:
Does this do any type coercion? E.g. date parsing? Was marshmallow doing that before?

If not, we should probably add that, but it can definitely be in a separate PR.